src/sinks/util/retries.rs

```diff
@@ -11,7 +11,10 @@ use futures::FutureExt;
 use tokio::time::{sleep, Sleep};
 use tower::{retry::Policy, timeout::error::Elapsed};
 
-use crate::Error;
+use crate::{
+    internal_events::{SinkRetryError, SinkRetryReasonError},
```

Comment:
Especially reading the usage here, I think the events can be collapsed into a single `SinkRetryError`.

Reply:
The thing is- the `SinkRetryError` contains an Error type, and the `SinkRetryReasonError` contains only a string (no Error type).
If I combined them, I'd have to make the args Option<..> , and then have a conditional in the `emit()` which would then call one or the other error! log.

Reply:
🤔 possibly convert the Error to a string via Display? Just seemed like an overly specific event to me.

Reply:
I guess I was trying to adhere to the original error! log messages. One specified "reason" , the other specified "error".

In the typical error! call, we see:

```
          error!(                                                                                                                                                                                                                                                                                                                                     
              message = %self.message,                                                                                                                                                                                                                                                                                                                
              error = %self.error,                                                                                                                                                                                                                                                                                                                    
...                                                                                                                                                                                                                                                                                             
          );        
```

Is it ok if the value self.error is not actually an error type ? :thought_balloon:  I guess it isn't covered in the instrumentation spec, I just seem to see it a lot so figured that was a requirement.

But if its not a requirement, and "error" in the error! call can be just a string, then I agree, these can be collapsed.

Reply:
I'd defer upward on these open questions 😄

Reply:
I'm starting to think that my inference about the error type requirement is invalid, that  you are right and that there are a some amount of events out there that fall under the category of overly specific.

Reply:
Addressed in https://github.com/vectordotdev/vector/pull/14225/commits/de926d09da37f5ec27f2f52ddc4e163e8336944a